<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Alexej Gossmann" />

<meta name="date" content="2016-10-19" />

<title>Basic usage of grpSLOPE</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Basic usage of grpSLOPE</h1>
<h4 class="author"><em>Alexej Gossmann</em></h4>
<h4 class="date"><em>2016-10-19</em></h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Group SLOPE (<em>gSLOPE</em>) is a penalized linear regression method that is used for adaptive selection of groups of significant predictors in a high-dimensional linear model. A unique property of the Group SLOPE method is that it offers group false discovery rate (<em>gFDR</em>) control (i.e., control of the expected proportion of irrelevant groups among the total number of groups selected by Group SLOPE). A detailed description of the method can be found in <a href="https://arxiv.org/abs/1610.04960">D. Brzyski, A. Gossmann, W. Su, and M. Bogdan (2016) <em>Group SLOPE — adaptive selection of groups of predictors</em></a>.</p>
<p>Group SLOPE is implemented in the R package <code>grpSLOPE</code>. As an introduction to the R package, in the following we will walk through a basic usage demonstration. First, we will simulate some data, before we feed it into <code>grpSLOPE</code>, and subsequently examine the output.</p>
</div>
<div id="data-generation" class="section level2">
<h2>Data generation</h2>
<p>We simulate a <span class="math inline">\(500 \times 500\)</span> SNP-data-like model matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">17082016</span>)

p     &lt;-<span class="st"> </span><span class="dv">500</span>
probs &lt;-<span class="st"> </span><span class="kw">runif</span>(p, <span class="fl">0.1</span>, <span class="fl">0.5</span>)
probs &lt;-<span class="st"> </span><span class="kw">t</span>(probs) %x%<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">1</span>,p,<span class="dv">2</span>)
X0    &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rbinom</span>(<span class="dv">2</span>*p*p, <span class="dv">1</span>, probs), p, <span class="dv">2</span>*p)
X     &lt;-<span class="st"> </span>X0 %*%<span class="st"> </span>(<span class="kw">diag</span>(p) %x%<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>))</code></pre></div>
<p>For example, the upper left <span class="math inline">\(10 \times 10\)</span> corner of <span class="math inline">\(X\)</span> looks as follows.</p>
<table style="width:28%;">
<colgroup>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
<col width="2%"></col>
</colgroup>
<tbody>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
</tr>
<tr class="odd">
<td>1</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>0</td>
</tr>
<tr class="even">
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>2</td>
</tr>
<tr class="odd">
<td>1</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
</tbody>
</table>
<p><strong>Note:</strong> In fact, with the default settings, the Group SLOPE method is guaranteed to control gFDR only when applied to a data matrix, where the columns corresponding to different groups of predictors are nearly uncorrelated. The relevant theoretical results can be found in <a href="https://arxiv.org/abs/1610.04960">Brzyski et. al. (2016)</a>. Only for the brevity of exposition we neither check for nor enforce low between-group correlations in this example.</p>
<p>We divide the 500 predictor variables into 100 groups of sizes ranging from 3 to 7.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">group &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>:<span class="dv">20</span>, <span class="dt">each=</span><span class="dv">3</span>),
           <span class="kw">rep</span>(<span class="dv">21</span>:<span class="dv">40</span>, <span class="dt">each=</span><span class="dv">4</span>),
           <span class="kw">rep</span>(<span class="dv">41</span>:<span class="dv">60</span>, <span class="dt">each=</span><span class="dv">5</span>),
           <span class="kw">rep</span>(<span class="dv">61</span>:<span class="dv">80</span>, <span class="dt">each=</span><span class="dv">6</span>),
           <span class="kw">rep</span>(<span class="dv">81</span>:<span class="dv">100</span>, <span class="dt">each=</span><span class="dv">7</span>))
group &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;grp&quot;</span>, group)
<span class="kw">str</span>(group)</code></pre></div>
<pre><code>##  chr [1:500] &quot;grp1&quot; &quot;grp1&quot; &quot;grp1&quot; &quot;grp2&quot; &quot;grp2&quot; &quot;grp2&quot; ...</code></pre>
<p>For further usage we keep additional information about the grouping structure of predictors, such as the total number of groups and the group sizes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># this generates a list containing a vector of indices for each group:</span>
group.id &lt;-<span class="st"> </span>grpSLOPE::<span class="kw">getGroupID</span>(group)
<span class="co"># this extracts the total number of groups:</span>
n.group &lt;-<span class="st"> </span><span class="kw">length</span>(group.id)
<span class="co"># this vector collects the sizes of every group of predictors:</span>
group.length &lt;-<span class="st"> </span><span class="kw">sapply</span>(group.id, <span class="dt">FUN=</span>length)
<span class="co"># this vector collects the group names:</span>
group.names &lt;-<span class="st"> </span><span class="kw">names</span>(group.id)</code></pre></div>
<p>In order to simulate a response variable, we randomly select 10 groups to be truly significant.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ind.relevant &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">sample</span>(<span class="dv">1</span>:n.group, <span class="dv">10</span>)) <span class="co"># indices of relevant groups</span></code></pre></div>
<p>The randomly selected truly significant groups are:</p>
<table style="width:82%;">
<colgroup>
<col width="6%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
<col width="8%"></col>
</colgroup>
<tbody>
<tr class="odd">
<td>grp4</td>
<td>grp13</td>
<td>grp22</td>
<td>grp34</td>
<td>grp35</td>
<td>grp59</td>
<td>grp65</td>
<td>grp68</td>
<td>grp72</td>
<td>grp78</td>
</tr>
</tbody>
</table>
<p>Then we generate the vector of regression coefficients, by sampling effect sizes for the significant groups from the Uniform(0,1) distribution.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">b &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, p)
for (j in ind.relevant) {
  b[group.id[[j]]] &lt;-<span class="st"> </span><span class="kw">runif</span>(group.length[j])
}</code></pre></div>
<p>Finally, we generate the response vector according to a linear model with i.i.d. <span class="math inline">\(\mathcal{N}(0, 1)\)</span> noise terms.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">y &lt;-<span class="st"> </span>X %*%<span class="st"> </span>b +<span class="st"> </span><span class="kw">rnorm</span>(p)</code></pre></div>
</div>
<div id="fitting-the-group-slope-model" class="section level2">
<h2>Fitting the Group SLOPE model</h2>
<p>We fit the Group SLOPE model to the simulated data. The function argument <code>fdr</code> signifies the target group-wise false discovery rate (gFDR) of the variable selection procedure.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(grpSLOPE)

model &lt;-<span class="st"> </span><span class="kw">grpSLOPE</span>(<span class="dt">X=</span>X, <span class="dt">y=</span>y, <span class="dt">group=</span>group, <span class="dt">fdr=</span><span class="fl">0.1</span>)</code></pre></div>
</div>
<div id="model-fit-results" class="section level2">
<h2>Model fit results</h2>
<p>The resulting object <code>model</code> of class “grpSLOPE” contains a lot of information about the resulting Group SLOPE model. Some of these parameters are shown below.</p>
<ul>
<li><p>Groups that were selected as significant by the Group SLOPE method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model$selected</code></pre></div>
<pre><code>##  [1] &quot;grp4&quot;  &quot;grp11&quot; &quot;grp13&quot; &quot;grp22&quot; &quot;grp34&quot; &quot;grp35&quot; &quot;grp59&quot; &quot;grp65&quot;
##  [9] &quot;grp68&quot; &quot;grp72&quot; &quot;grp78&quot;</code></pre>
<p>Notice that the model has correctly identified all significant groups, and additionally has falsely reported the group “grp11” as significant.</p></li>
<li><p>The estimated noise level <span class="math inline">\(\hat{\sigma}\)</span> (true <span class="math inline">\(\sigma\)</span> is equal to one).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sigma</span>(model) <span class="co"># or equivalently: model$sigma</span></code></pre></div>
<pre><code>## [1] 0.9886372</code></pre></li>
<li><p>The regression coefficients, which can be displayed either on the normalized scale (i.e., the scale corresponding to the normalized versions of <span class="math inline">\(X\)</span> and <span class="math inline">\(y\)</span>, on which all the parameter estimates are computed internally),</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># the first 13 coefficient estimates</span>
<span class="kw">coef</span>(model)[<span class="dv">1</span>:<span class="dv">13</span>]</code></pre></div>
<pre><code>##   grp1_1   grp1_2   grp1_3   grp2_1   grp2_2   grp2_3   grp3_1   grp3_2 
## 0.000000 0.000000 0.000000 0.000000 0.000000 0.000000 0.000000 0.000000 
##   grp3_3   grp4_1   grp4_2   grp4_3   grp5_1 
## 0.000000 4.127673 1.613667 5.232564 0.000000</code></pre>
<p>or on the original scale of <span class="math inline">\(X\)</span> and <span class="math inline">\(y\)</span>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># intercept and the first 13 coefficient estimates</span>
<span class="kw">coef</span>(model, <span class="dt">scaled =</span> <span class="ot">FALSE</span>)[<span class="dv">1</span>:<span class="dv">14</span>]</code></pre></div>
<pre><code>## (Intercept)      grp1_1      grp1_2      grp1_3      grp2_1      grp2_2 
##   3.8190431   0.0000000   0.0000000   0.0000000   0.0000000   0.0000000 
##      grp2_3      grp3_1      grp3_2      grp3_3      grp4_1      grp4_2 
##   0.0000000   0.0000000   0.0000000   0.0000000   0.2727357   0.1000815 
##      grp4_3      grp5_1 
##   0.4077402   0.0000000</code></pre>
<p>(notice that the coefficients are named corresponding to the given grouping structure). As expected from a penalized regression method, we observe some shrinkage, when we compare the above to the true parameters.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># true first 13 coefficients</span>
b[<span class="dv">1</span>:<span class="dv">13</span>]</code></pre></div>
<pre><code>##  [1] 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000
##  [8] 0.0000000 0.0000000 0.3677210 0.1713692 0.6425599 0.0000000</code></pre></li>
<li><p>It might also be interesting to plot the first few elements of the regularizing sequence <span class="math inline">\(\lambda\)</span> used by the Group SLOPE method for the given inputs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(model$lambda[<span class="dv">1</span>:<span class="dv">10</span>], <span class="dt">xlab =</span> <span class="st">&quot;Index&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Lambda&quot;</span>, <span class="dt">type=</span><span class="st">&quot;l&quot;</span>)</code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC9FBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////rLY7VAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAN2UlEQVR4nO2de1wVVR7AD4KXQOVKEPLwkW9t1TaNrNXMB9vD1bIHrWlbSWJkGoV03RXLbLXFXExFso1QfIG2tqG7mYmV5laUuK5pWmZlruUDlQwSmPPPzly4coGZ85u5c+bM+PH3/fiZez9zf/M7h69zZ86dOecMoQgTYncFnA4KAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBcBb09KBLhKHHRQo6MCW1AfeLZZcGPfeIFHQsz0foEh75BDBAqKBG2i7nm88y7BX0zja+WS3AXkHr7uCb1QLsFVQZcZpvWv7YfAy6s5BvWv7YLOj1e/im5Y/NgircP/HNyx27T/O/3cA3L3fsFpQ7gW9e7tgt6HjkL3wT88ZuQXToZr6JeWO7oIWP8k3MG7GCfnrHR9grDauORNfwyGwZYgV9lOQjeIFv3aBSHpktw/avGH1hGt/MnLFf0BcJEt/UfLFfEL3m33xT88UBgmY9wzc1Xxwg6LOufFPzxQGCaHe9dbADJwiaMZtvbq44QdDO/nxzc8UJgqSOB/gm54kTBNGp8/gm50nAgsqzTZXbRNDWRFO5LMW4oLrCzAyZQfGmym0iqDbmG1PJrMS4oCzSP7RDYpvIt+FtKj7XvBrW9M7qpBydtRCPcUFd0uiqMfTHnv9kRb951+2FdJmLuGbX+q8ujfQR9KL/+pJhuissGuOCWq+mx2IkmjOEEbyWJI5pndFu7mZP8DL/9dJpH033oOr2/9NdY8EYFxQrN+s67KEFbRjB106U6BqinJueGKAe0azzwvg8ndUQjnFBqZHL6ehJ347tyQgOK5QPQGS7/G5NuHpEM0HFt+mshnCMCzqbMo6Wh5Pg1Yzgnh5K3yOKgzm91COaCTrvdupN+gDbQWe2HGYFvxg8ZVbM0LgtJ9ZH/FE9onn/oHErdNZDNNa0pGuy4qOfklIJIb/TuLXcXNCKcUbyC8SgoCg/2PGS/G97QZnW5dTmghx7k96goNzc3EXuThkLM7t03QRsIJ09xbjY3KIL3m3FOisiGONfscd/UyUvq4eksaIPeXqEEuLq9syXGgEtBOWN11kRwRgXlJDvfcln/Rb7OCI+NWdFwaLHu7g/VY9oIeiH9lU6ayIW44Li5npf5nRiBA+/pbL+TfVdI9UjWvZyHVaisyZiMS4oJXx9Ha0rDpvCCI541ffuLbd6REtBOZN01kQsxgWdG07a9W5HklhnncQJvuPzUzeoR7QU9J0zb9IH0A6SSrMzFu5gBhcFjVq+8/P9H742plWTk9M+30iE1NZ/bbFR4rs6qyKUQBqKP5au3XWOHV0ygngZ0fTAcrxxKMLSFtvMm6qzKkIxLqhmemv5L3fPB+6on9yzZUv5Sc2PVYYiHIyt01kXkRgXNItk7j1V/gR5WddmlRXq69XGavT7UGddRGJcUPd078vkX+nabJTGbzg1QbNn6KyLSIwLurL+sLsyUtdmC1LU16sJKr9aZ11EYlzQPY8oS+n3Y02VqzocqtduUzktwaCg3bt3b425e8OuDXd22scMrzl83vv6k8bFZlVBz8zSWRmBGBRE/GAE1z4XRlzpF+R32QaOQXSXvuOaUAwKOuQHI3hR0PTi9GClE70hQVKn/TprI44AGorVB/d7YQRfo7T58skbBgXRaS/orI04jAta1wb+ioWtUZb3dqsyKGjb9TprIw7jgjrfsGGvF0Zwb29b6bv2kyRjgmpjmPcC7MC4IPe/4OAFQakbf5Z3NpKSZkgQfXShzuoIw7igsTqOEzXz2hPlGFUU0/SbeK5xKIL6rdTNN+qsjjCMCzre+cl1f1dghktHvVdQf9n6iv/asotDEUIWqG5Wc91rOusjCuOClupoB8FoTSywz2ldhYwL6jpuX40XU+Vqzrzw51HOGpkQwEX7LTzK1RRUOziXR35uGBc0PZ1Hudpzd+yPZrXRhWNcUH7H25+bp2CqXMbkJguGOOnKYgAXzHyYKpchqG7YIlOp+eKIftLNOBzzBd/CzOCIftLNefmmWsanYnFEP+nmSLf+xVRynljaT5oBewaqr6NZP4WFYlE/aRBgiq68gRdMpeeHNf2kYQBB0h1zTaXnhzX9pGGgSd6+jykzlZ8b1vST1mKbxlAEFfKvdcasJ9b0k9akQn0oghp3ZwVSAHes6ScNAws61uFjc0XwIeCGYpnHVLk6Jppc3dcJvRYDFpRvzQUzf+4z93/ABycLOhH3gakyuOBkQXRj759NFcIDRwui458yVQgPnC2ootN2U6VwwKCgIRfpIUIQfatrpalizGNQ0G1+mCpX73zSD9nd9dWJVxT9qehs800Opwuih/o/aOupzPGCaGXywCN8yzaEdYLUBtT95/7kBkJaDkXQJC/OxkEKFgnSGFB3osjHFUYOLdvj5xsqnSfWCApgQB2TbxMfOG8kniPWCApkQB2Tqod/bVPfM2sEBTKgDiAvlkunCcNYIyiQAXUQ7ydAA4wswRpBWgPqGgngySxHB99vw9h6i85iGgPqGgnk0TXVKQO0Rplbh2XtIOMD6nSQF70xkM3MILah2EiADz/aEe8R3HlIbEOxkUCfDvX9TWPPBLZlgFwaDUU/qh+LnflVgNsGwiXSUPTnoCcmqUhY34ZLpqHoT3VRUpxHUMtabEPxTLHvx2rYMpXNDHDAc1VSkYipGsQ2FD/zXe1IDl9pJJ8a8m4U77H+SpFdDcUbdxnKp85+T7Tlu5FdDUUuguRf+UVJCR5LR3dYfclVa+YFToJkPk2NiL8lZd76cmuuGFktSGvmBX6CZE6XFXiSB7WNHJQ8u6gMmHbFKFYL0pp5gaugeqRv3l02465rrug4fLKHB/X9/e26q2GBoAbqjmzNnc+D+qkvrRIEzbxgnSDOWCMInnnhMhcEz7xw48w8NRYnTYS4dzQYMvoeMGTUeChiwmhvjRIsEQTPvLA8VZX7XH0h4t1gSPs4MMTVDYroFeKt0fSzVgiCZ17Q4FAPMCT/YTDkUfiXcB9wlpAT0WCSJhj6Q+GZFzS4XATBMy9ocLkI0pp5AeRyEaQ18wLI5SMoQFAQAAoCQEEAX/cBQwrhZx8/9joY0g+c3qEiFkzSBDGCqPY1SB8X4KbtOXgsHlyOnhB/BAm6dEFBACgIAAUBoCAAFASAggBQEAAKAkBBACgIAAUBiBBU+1Lf8H6LofnL6kZOYH6+d+yVnecwOwvXZPcKv24dI2BGhveleLB75CdAbS4iQtC8oOmr04KeA6KWEqagA9GjX00jzAFms4MzV04gb2l+fuhKr6CSVg8sv9mtd9pLAYKkyMnycloYu8/Y4bZtmYImD5N3wYx7WSHxU+TCrtd67OR7Q0OIV9CIkRI9F6d3AhEBgo4SZaBBMWH2X60b8eAQlqDqdqso8CWVopXHU4y5XePjvdnZ3j3oJFFuOKTC1zjrESCoar8yvjndxewqtiT2FFPQV2TVza6EZ5l7YXqH98+sbc3oRtpdEbSH7JSXOaE6B2GJOovlBz/J+vhwm42UKWgHCZ+xKSski5Wkbigh5GlGgFfQFvK5vFxBdN6cFyPo6H3kAdb10roRshymoBKiTIiSHs7YhaQpsXmlL7Rj9NRuEKRc2C8gOi+9ChFU1L7bG8yAv0V+WVExOLlCeyTCR+QdefkPclA7yXtEmSRzrkt7VIxXUDlRHue1yOWgr1gxmQTMypXRMF289oT633pP3xvIMe0kBeSEvNzs/Qap073+IJ0vL6d2Y9foIgIE/RKTAv1vfVkq0y+p9IRmhDRAeaLtxI6MTJ+QDfJypkv7u+wVRIePkU+KV2cCVfIhQNC7JC1bARqZyjwG0VXkD4WTSCEjQrojYu7aacFZ2hH1gkqCni4Z59Y7ClSAoFcavj8a/T4vwhZEVya2Hch+cHTlzJ5h/ZYwWkv1gui66yNG6J4QFH+sAqAgABQEgIIAUBAACgJAQQAoCAAFAaAgABQEgIIAUBAACgJAQQAoCAAFAaAgABQEgIIAUBAACgJAQQAoCAAFAaAgAOcJym1xjzo3ypaK1IOCAFAQgDMFdVmZ3Tci+RSl5yYnJExdJAuSXh0Y3n8NpduUnmbFIeXCquNQQUPu3LTY9TiVhoc/n3+TWxb0UtATayYq3YMe6Vl1Nv5P4qrjUEF9ail96AZaStZTWtUpila6lY7f9/eg9ORVzz7ZR+BzNh0qSOkg5xlEs9spgzMyouhH5L/ymzdbXaB0lStkp8DqOFSQ8mB6WVBGb2XNwii6vqGX2hFKL0T3ETn/vUMFZVOvoIXePSgziu4gb+9WOE/p/A6hZmb1NoqzBX2gHIOqu0TR02GvyWuWPSLRL0LXZUX+IK46zhYkjQp/vmBoV/ksNit8dlFmq+dp3c23Sj93fVBcdZwtiFZO7hiXtklpB+X0C+uTI9GloYeUcQnbhFXHeYIcBgoCQEEAKAgABQGgIAAUBICCAFAQAAoCQEEAKAgABQGgIAAUBICCAFAQAAoCQEEAKAgABQGgIAAUBICCAFAQwP8Be7rTg8rlGcQAAAAASUVORK5CYII=" /><!-- --></p></li>
</ul>
<p>We can further check the performance of the method by computing the resulting group false discovery proportion (gFDP) and power.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">true.relevant &lt;-<span class="st"> </span>group.names[ind.relevant]
truepos       &lt;-<span class="st"> </span><span class="kw">intersect</span>(model$selected, true.relevant)

n.truepos  &lt;-<span class="st"> </span><span class="kw">length</span>(truepos)
n.selected &lt;-<span class="st"> </span><span class="kw">length</span>(model$selected)
n.falsepos &lt;-<span class="st"> </span>n.selected -<span class="st"> </span>n.truepos

gFDP &lt;-<span class="st"> </span>n.falsepos /<span class="st"> </span><span class="kw">max</span>(<span class="dv">1</span>, n.selected)
pow &lt;-<span class="st"> </span>n.truepos /<span class="st"> </span><span class="kw">length</span>(true.relevant)

<span class="kw">print</span>(<span class="kw">paste</span>(<span class="st">&quot;gFDP =&quot;</span>, gFDP))</code></pre></div>
<p>[1] “gFDP = 0.0909090909090909”</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">paste</span>(<span class="st">&quot;Power =&quot;</span>, pow))</code></pre></div>
<p>[1] “Power = 1”</p>
<p>We see that the method indeed did not exceed the target gFDR, while maintaining a high power.</p>
</div>
<div id="lambda-sequences" class="section level2">
<h2>Lambda sequences</h2>
<p>Multiple ways to select the regularizing sequence <span class="math inline">\(\lambda\)</span> are available.</p>
<p>If a group structure with little correlation between groups can be assumed (i.e., groups in the standardized model matrix are nearly orthogonal), then we suggest to use the sequence “corrected”, which is the default.</p>
<p>The <span class="math inline">\(\lambda\)</span> sequences “mean” and “max” can be used together with the options <code>orthogonalize = FALSE</code> and <code>normalize = FALSE</code>, when the columns of the model matrix are exactly orthogonal to each other (“max” is more conservative, giving exact gFDR control only when all groups have the same size, and otherwise resulting in a lower gFDR than the target level).</p>
<p>Alternatively, any non-increasing sequence of appropriate length can be utilized. However, we do not recommend to use any other <span class="math inline">\(\lambda\)</span> sequences unless you really know what you are doing.</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<ul>
<li><p><a href="https://arxiv.org/pdf/1407.3824v2.pdf">Bogdan, M., van den Berg, E., Sabatti, C., Su, W., and Candès, E. J. (2015). <em>SLOPE — Adaptive Variable Selection via Convex Optimization.</em></a> The Annals of Applied Statistics, vol. 9, no. 3, p. 1103.</p></li>
<li><p><a href="https://arxiv.org/abs/1610.04960">Brzyski, D., Gossmann, A., Su, W., and Bogdan, M. (2016). <em>Group SLOPE — adaptive selection of groups of predictors</em></a> (submitted to the Journal of the Royal Statistical Society: Series B).</p></li>
<li><p><a href="http://dx.doi.org/10.1145/2808719.2808743">Gossmann, A., Cao, S., and Wang, Y.-P. (2015). <em>Identification of Significant Genetic Variants via SLOPE, and Its Extension to Group SLOPE.</em></a> In Proceedings of the 6th ACM Conference on Bioinformatics, Computational Biology and Health Informatics, BCB ’15 (pp. 232–240). New York, NY, USA: ACM.</p></li>
</ul>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
